autoload/lsp.vim

```diff
@@ -169,6 +181,7 @@ function! lsp#register_server(server_info) abort
         \ }
     call lsp#log('lsp#register_server', 'server registered', l:server_name)
     doautocmd <nomodeline> User lsp_register_server
+    let s:servers_update_cnt += 1
```

Comment:
unregister_server should also increment

Reply:
also what about if server exists?

Reply:
I cannot find any function do something like unregister_server.
There's a `lsp_unregister_server` autocmd called by `unregister_events`, but the server is not removed from `s:servers`.
The original `get_allowed_server` just get all s:servers and does not handle server exit event.
I think it'd better solve these problems in other merge requests, since this merge request is 'improve perf'.